### PR TITLE
Handle secondary GitHub ratelimits

### LINKED ISSUE #1911: GitHub `visibilityOf` function doesn't handle ratelimits
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### TruffleHog Version

The latest one at time of writing this, built locally.

### Trace Output

N/A 

### Expected Behavior

When the API ratelimit is encountered, TruffleHog should wait and retry. 

### Actual Behavior

TruffleHog proceeds without waiting, spitting out numerous ratelimiting errors.

### Steps to Reproduce

To reproduce this issue, start a scan that includes issue comments against a large number of repositories.

```sh
./trufflehog github --org coinbase --issue-comments --token=github_pat_realtoken --pr-comments --only-verified
🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷

2023-10-18T12:38:47-04:00       info-0  trufflehog      Completed enumeration   {"num_repos": 109, "num_orgs": 1, "num_members": 0}
2023-10-18T12:38:47-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/code-of-conduct.git"}
2023-10-18T12:38:47-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/coinbase-android-sdk.git"}
2023-10-18T12:38:48-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/bifrost.git"}
...
2023-10-18T12:38:56-04:00       info-0  trufflehog      scanning github pull request descriptions       {"repository": "https://github.com/coinbase/coinbase-ubercart.git"}
2023-10-18T12:38:56-04:00       error   trufflehog      Could not get Github repository {"repo": "https://github.com/coinbase/coinbase-zencart.git", "error": "GET https://api.github.com/repos/coinbase/coinbase-zencart: 403 You have exceeded a secondary rate limit. Please wait a few minutes before you try again. If you reach out to GitHub Support for help, please include the request ID E048:60D6:14AD678:2A49754:65300A21."}
2023-10-18T12:38:56-04:00       error   trufflehog      Could not get Github repository {"repo": "https://github.com/coinbase/custom-salus-example.git", "error": "GET https://api.github.com/repos/coinbase/custom-salus-example: 403 You have exceeded a secondary rate limit. Please wait a few minutes before you try again. If you reach out to GitHub Support for help, please include the request ID E048:60D6:14AD67C:2A4976C:65300A21."}
...
```

## Environment

N/A

## Additional Context

N/A

### References

N/A

### PR DESCRIPTION
**Pull request by rgmz**
### Description:

This fixes #1911, which turned out to be caused by [GitHub's secondary rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#secondary-rate-limits).

```sh
# debug print statements removed in the final commit
$ ./trufflehog github --org coinbase --issue-comments --token=github_pat_xxx --only-verified
🐷🔑🐷  TruffleHog. Unearth your secrets. 🐷🔑🐷

2023-10-18T14:06:11-04:00       info-0  trufflehog      Completed enumeration   {"num_repos": 109, "num_orgs": 1, "num_members": 0}
2023-10-18T14:06:12-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/code-of-conduct.git"}
2023-10-18T14:06:12-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/.github.git"}
...
Waiting for: 1m0s
Waiting for: 1m0s
2023-10-18T14:06:25-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/rosetta-ethereum.git"}
Waiting for: 1m0s
2023-10-18T14:06:25-04:00       info-0  trufflehog      scanning github issue descriptions      {"repository": "https://github.com/coinbase/coinbase-pro-node.git"}
Waiting for: 1m0s
2023-10-18T14:06:25-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/rosetta-cli.git"}
Waiting for: 1m0s
Waiting for: 1m0s
Waiting for: 1m0s
Waiting for: 1m0s
2023-10-18T14:07:05-04:00       info-0  trufflehog      scanning github issue comments  {"repository": "https://github.com/coinbase/salus.git"}
2023-10-18T14:07:12-04:00       info-0  trufflehog      scanning github issue descriptions      {"repository": "https://github.com/coinbase/salus.git"}
...
```

It also contains a minor change where, instead of calling `s.visibilityOf()` in the individual processing methods, it is called once in `processRepoComments()` and passed that along in `repoInfo`.

(However, this doesn't actually reduce the number of API calls because `visibilityOf` caches results.)

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/sources/github/github.go [hunk 1]
```diff
@@ -1131,20 +1144,30 @@ var (
...
+	owner      string
+	repo       string
+	repoPath   string
+	visibility source_metadatapb.Visibility
 }
 
 func (s *Source) processRepoComments(ctx context.Context, repoPath string, trimmedURL []string, repoURL *url.URL, chunksChan chan *sources.Chunk) error {
+	if !(s.includeIssueComments || s.includePRComments) {
+		return nil
+	}
```
Comment (rosecodym) [inline pkg/sources/github/github.go:1156]: Is the idea here that the issue or PR initial description constitutes a "comment" from a configuration standpoint?

**Review by rosecodym — COMMENTED** [review]
Thanks for this! Please let me know if I understand the change correctly. I read two major parts:
* Handling of Github's secondary rate limit has been added (which entailed some refactoring of the rate limit handling logic generally).
* A repo's visibility is now retrieved only once and simply passed into the various `chunk*` functions, to minimize the chance of hitting this secondary rate limit.

File: pkg/sources/github/github.go (hunk 1 above)
Comment (rgmz) [reply pkg/sources/github/github.go:1156]: The existing code implies such, which I think makes sense. 

This guard will exit early and prevent needless calls to `s.visibilityOf`. Alternatively, we could wrap the [`s.processRepoComments` call](https://github.com/trufflesecurity/trufflehog/blob/20eb89d089095425c0feae01f78ddadb400c2839/pkg/sources/github/github.go#L1088) in this condition.

**Comment by rosecodym** [comment]
Neat, thanks! (And I learned about comment footnotes!) We can merge this in after the merge conflicts are resolved.

**Comment by rgmz** [comment]
Should be resolved now.

**Review by rosecodym — APPROVED** [review]

**Pull request merged**